system/Images/Image.php

```diff
@@ -97,7 +98,7 @@ public function copy(string $targetPath, string $targetName = null, int $perms =
 
 		if (empty($targetName))
 		{
-			throw ImageException::forInvalidFile($targetName);
+			throw FileException::forInvalidFilename($targetName);
```

Comment:
The ImageException class extends the FrameworkException (just like the FileException class). The point of using ImageException here was that you could focus on catching one type of exception, instead of needing to add multiple catch blocks which is messy before 7.2.

Reply:
FrameworkException ... forInvalidFile throws new self(...), which is a FrameworkException, and not an ImageException.
I am modifying FrameworkException to throw new static(...) instead, which has the behavior you describe.